lib/daemon_runner/semaphore.rb

```diff
@@ -28,7 +28,7 @@ def lock(name, limit = 3, **options)
         logger.debug e.backtrace.join("\n")
         raise
       ensure
-        lock_thr.kill
+        lock_thr.kill unless lock_thr.nil?
```

Comment:
Something like this?
```ruby
unless lock_thr.nil?
  lock_thr.kill
  semaphore.release
end
```
That seems like it would be closer to [the original behavior](https://github.com/rapid7/daemon_runner/pull/32/files).